tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -59,12 +54,24 @@ namespace {
 // Performs lowering to TOSA dialect.
 class LegalizeTFL : public TosaLegalizeTFLPassBase<LegalizeTFL> {
  public:
-  explicit LegalizeTFL() {}
+  explicit LegalizeTFL(std::unordered_set<std::string> &legalization_disable) {this->legalization_disable = legalization_disable;}
   void runOnFunction() override;
+ private:
+  std::unordered_set<std::string> legalization_disable;
```

Comment:
Please make this a pass option declared and documented in the .td file.

The passes must be hermetic: they should not have data member that aren't Options as these won't be serialized and produce correct reproducer on failures.

In the documentation for the option (and the public C++API), please include example of the names, users shouldn't have to read the code implementation to figure it out.

Finally, please add a test for the command line option.

Reply:
1. I have expanded the comment in the td file
2. I have added a comment in the passes.h file

3. On adding a test, I am not sure how to achieve this with the current infrastructure as it uses tf-opt to achieve the testing and not sure how to get options loaded (a legalizatin-disable function), without writing say tosa-opt which is not a trivial piece of work. 
4. On adding the pass option to the td file I am not sure how to achieve this using tablegen. Can you point me to an example please ?

Reply:
Here is an example of a "complex type" being serialized as a string option: https://github.com/llvm/llvm-project/commit/32fe1a8a25944d0c8ae1632684e912f1fb147253

You likely want a `ListOption` (see doc: https://mlir.llvm.org/docs/PassManagement/#instance-specific-pass-options )

Reply:
Actually there is an almost exact example of what you want in `mlir/lib/Rewrite/FrozenRewritePatternSet.cpp`.

The option are defined in `mlir/include/mlir/Rewrite/PassUtil.td` 

I think you can model this and have the API take a `ArrayRef<std::string>` for the pass creation, locally use a `DenseSet<StringRef>`.

Reply:
Oh and when you have an option, you should be able to write a test with tf-opt :)
It'll look something like this:
`tf-opt --pass-pipeline="func(tf-tosa-legalization{keep-tfl-ops=TFLAdd,TFLConv2D})"'

Reply:
@joker-eph thanks for the pointers above and examples.
I have uploaded a patch that should address this.